### ISSUE TITLE
Not-in-scope variable "shadows" in-scope variable

### ISSUE THREAD

**Issue by anuyts** [agda/agda#5676]
This bug occurs in Agda 2.6.0.1, I cannot immediately check if it occurs in a more recent version.

For the following code:
```agda
data Sing (A : Set) : Set where
  sing : A → Sing A

data IsSing : Set → Set₁ where
  isSing : {A : Set} → IsSing (Sing A)

f : {X : Set} → IsSing X → Set
f {A} isSing = {!!}
```
the information we get for the hole is
```
Goal: Set
————————————————————————————————————————————————————————————
A : Set
A : Set  (not in scope)
```
i.e. Agda has assigned the argument to `isSing` the same name `A` that the user has used for the first argument to `f`. This can clearly cause misleading type information when programming further.

**Comment by nad** [comment]
With a more recent version I get the following output:
```
Goal: Set
————————————————————————————————————————————————————————————
A  : Set
A  = Sing A₁
A₁ : Set   (not in scope)
```

**Comment by anuyts** [comment]
Then we're good.

**Comment by andreasabel** [comment]
This is already fixed in 2.6.1.
